docs/getting_started/howto/plugin_system.rst

```diff
@@ -4,88 +4,107 @@ Plugin system
+:code:`installable_forward_model_steps` which gives a lot more control.
 
 .. code-block:: python
 
-   from ert.shared.plugins.plugin_response import plugin_response
-   @hook_implementation
-   @plugin_response(plugin_name="ert")
-   def installable_jobs():
-      return {"SOME_JOB": "/path/to/some/job/config"}
+   import ert
 
-This way you only need to return the data of interest, which is specified in the hook specification.
+   @ert.plugin(name="my_plugin")
+   def installable_jobs() -> dict[str, str]:
+       return {
+                "job_name": "/path/to/workflow.config"
+        }
 
-Hook specifications
--------------------
 
-The following hook specifications are available to use:
+    class MyForwardModel(ert.ForwardModelStepPlugin):
+        def __init__(self):
+            super().__init__(
+                name="MY_FORWARD_MODEL",
+                command=["my_executable", "<parameter1>", "<parameter2>"],
+            )
 
-Forward models
-~~~~~~~~~~~~~~
-To install forward models that you want to have available in ERT you can use the following hook specification:
+        def validate_pre_realization_run(
+            self, fm_step_json: ert.ForwardModelStepJSON
+        ) -> ForwardModelStepJSON:
+            if fm_json["argList"][0] not in ["start", "stop"]:
+                raise ert.ForwardModelStepValidationError(
+                    "First argument to MY_FORWARD_MODEL must be either start or stop"
+                )
+            return fm_step_json
 
-.. code-block:: python
+        def validate_pre_experiment(self, fm_step_json: ert.ForwardModelStepJSON) -> None:
+            pass
+
+        @staticmethod
+        def documentation() -> Optional[ert.ForwardModelStepDocumentation]:
+            return ert.ForwardModelStepDocumentation(
+                category="utility.templating",
+                source_package="my_plugin",
+                source_function_name="MyForwardModel",
+                description="my plugin description",
+            )
 
-   @hook_specification
-   def installable_jobs():
-      """
-      :return: dict with job names as keys and path to config as value
-      :rtype: PluginResponse with data as dict[str,str]
-      """
 
+   @ert.plugin(name="my_plugin")
+   def installable_forward_model_steps() -> list[ert.ForwardModelStepPlugin]:
+       return [MyForwardModel]
+
+
+Notice that by using :code:`installable_forward_model_steps`, validation can be added
+where the methods ``validate_pre_experiment`` or ``validate_pre_realization_run`` can
+throw ``ForwardModelStepValidationError`` to indicate that the configuration of the
+forward model step is invalid.
 
 
 Forward model documentation
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-To provide documentation for a forward model, the following hook_specification can be used.
+To provide documentation for a forward model, use the :code:`job_documentation` name.
 If you are the plugin that provided the job with the name :code:`job_name`,
```

Comment:
This section corresponds more to the section above the new forward model step documentation unless it pertains to the one just below it (workflow jobs)